core/app/[locale]/(default)/account/(tabs)/orders/page-data.ts

```diff
@@ -0,0 +1,369 @@
+interface CustomerOrdersArgs {
+  after?: string;
+  before?: string;
+  limit?: number;
+  filterByStatus?: OrderStatus;
+  filterByDateRange?: OrderDateRange;
+}
+
+export const getCustomerOrders = cache(
+  async ({
+    before = '',
+    after = '',
+    filterByStatus,
+    filterByDateRange,
+    limit = 2,
+  }: CustomerOrdersArgs) => {
+    const customerAccessToken = await getSessionCustomerAccessToken();
+    const paginationArgs = before ? { last: limit, before } : { first: limit, after };
+    const filtersArgs = {
+      filters: {
+        ...(filterByDateRange && { dateRange: filterByDateRange }),
+        ...(filterByStatus && { status: filterByStatus }),
+      },
+    };
+    const response = await client.fetch({
+      document: CustomerAllOrders,
+      variables: { ...paginationArgs, ...filtersArgs },
+      customerAccessToken,
+      fetchOptions: { cache: 'no-store' },
+    });
+
+    const orders = response.data.customer?.orders;
+
+    if (!orders) {
+      return undefined;
+    }
+
+    const data = {
+      orders: await Promise.all(
+        removeEdgesAndNodes(orders).map(async (order) => {
+          const shipping =
+            order.consignments?.shipping &&
+            (await addProductAttributesToShippingConsignments(order.consignments.shipping));
+
+          return {
+            ...order,
+            consignments: {
+              shipping,
+            },
+          };
+        }),
+      ),
+      pageInfo: orders.pageInfo,
+    };
+
+    return data;
+  },
+);
+
+interface OrderDetailsProps extends CustomerOrdersArgs {
+  orderId: string;
+}
+
+export const getOrderDetails = cache(
+  async ({ orderId, before = '', after = '', limit = 2 }: OrderDetailsProps) => {
+    const customerAccessToken = await getSessionCustomerAccessToken();
+    const paginationArgs = before ? { last: limit, before } : { first: limit, after };
+
+    const response = await client.fetch({
+      document: CustomerOrderDetails,
+      variables: {
+        ...paginationArgs,
+        filter: {
+          entityId: +orderId,
+        },
+      },
+      fetchOptions: { cache: 'no-store' },
+      customerAccessToken,
+    });
+    const order = response.data.site.order;
+
+    if (!order) {
+      return undefined;
+    }
+
+    const shipping =
+      order.consignments?.shipping &&
+      (await addProductAttributesToShippingConsignments(order.consignments.shipping));
+
+    const data = {
+      orderState: {
+        orderId: order.entityId,
+        status: order.status,
+        orderDate: order.orderedAt,
+      },
+      summaryInfo: {
+        subtotal: order.subTotal,
+        discounts: order.discounts,
+        shipping: order.shippingCostTotal,
+        tax: order.taxTotal,
+        grandTotal: order.totalIncTax,
+      },
+      paymentInfo: {
+        billingAddress: order.billingAddress,
+        // TODO: add payments data
+      },
+      consignments: {
+        shipping,
+      },
+    };
+
+    return data;
```

Comment:
🍹 We should probably do all the remapping in application code rather than at the boundary.